test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -894,39 +894,109 @@ public async Task VerifyRestorePackageByRestoreNuGetPackagesContextMenu(ProjectT
 
             SolutionService solutionService = VisualStudio.Get<SolutionService>();
             solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
-            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ClassLibrary, ProjectTargetFramework.V48, "TestProject");
+            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, "TestProject");
             VisualStudio.ClearOutputWindow();
             solutionService.SaveAll();
 
             CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
             NuGetUIProjectTestExtension uiwindow = nugetTestService.GetUIWindowfromProject(project);
             uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
 
-            var packageFolderPath = _pathContext.PackagesV2;
-            var installedPackageFolderPath = Path.Combine(packageFolderPath, "Contoso.A.1.0.0");
+            if (projectTemplate.Equals(ProjectTemplate.NetCoreConsoleApp))
+            {
+                packageFolderPath = _pathContext.UserPackagesFolder;
+                installedPackageFolderPath = Path.Combine(packageFolderPath, TestPackageName);
+            }
+            else
+            {
+                packageFolderPath = _pathContext.PackagesV2;
+                installedPackageFolderPath = Path.Combine(packageFolderPath, "Contoso.A.1.0.0");
+            }
+
             Directory.Exists(installedPackageFolderPath).Should().BeTrue();
 
-            // Act
+            // Act (we doesn't edit the "nuget.config" file by setting the "automatic" to "true" since it is enabled by default)
             Directory.Delete(packageFolderPath, true);
             Directory.Exists(packageFolderPath).Should().BeFalse();
             solutionService.Build();
 
             // Assert
             Directory.Exists(installedPackageFolderPath).Should().BeTrue();
         }
+
+
+        [DataTestMethod]
+        [Timeout(DefaultTimeout)]
+        public void VerifyPackageNotRestoredAfterDisablingPackageRestore()
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+
+            SolutionService solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.IOSLibraryApp, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            _pathContext.Settings.DisableAutoRestore();
+            CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
+            VisualStudio.ObjectModel.Shell.ToolWindows.ErrorHub.ShowErrors();
+
+            // Assert
+            VisualStudio.AssertErrorListContainsSpecificError("NuGet restore is currently disabled.");
```

Comment:
Test failed here, where the CI run had an empty error list.

I wonder if it's reading the error list before the restore task has completed?

> Expected collection not to be empty because Empty errors in error list.

Reply:
The test failed in the original code because the list of errors was not displayed on vs as below screenshot, so we add `VisualStudio.ObjectModel.Shell.ToolWindows.ErrorHub.ShowErrors();` and now the failure never occurs.
![MicrosoftTeams-image (2)](https://github.com/NuGet/NuGet.Client/assets/84695155/c788176f-9667-49ea-b2cf-4815a9451dc0)